pkgs/development/node-packages/overrides.nix

```
@@ -597,6 +597,11 @@ final: prev: {
 
   wrangler = prev.wrangler.override (oldAttrs: {
     meta = oldAttrs.meta // { broken = before "16.13"; };
+    buildInputs = with pkgs.llvmPackages; [ libraries.libcxx libunwind ] ++ lib.optional stdenv.isLinux pkgs.autoPatchelfHook;
```

```suggestion
    buildInputs = [ pkgs.llvmPackages.libcxx pkgs.llvmPackages.libunwind ] ++ lib.optional stdenv.isLinux pkgs.autoPatchelfHook;
```

https://nix.dev/recipes/best-practices#with-scopes

---

greetings @wegank  !

Appreciate your time, especially since you've reviewed/merged many of my MRs

I am happy to make a change, but I just want to understand better what is reasoning for such change.

From the link that you have provided I understand

- `with (import <nixpkgs> {}); is bad`, since it add a lot of things into scope, but should not be an issue here
- `When more than one with used, it’s not clear anymore where the names are coming from.`, true, but in here `with` is scoped to `buildInputs`
- `Scoping rules for with are not intuitive, see this [Nix issue for details](https://github.com/NixOS/nix/issues/490).`  Yes, I am aware that `with` doesn't shadow local variables

which of those is concern ? or please specify 

again, appreciate your time, my questions are merely for educational purpose

---

I've applied suggestion, but would appreciate an explanation ( no matter how brief )

---

ah yes, that makes total sense, actually I used `pkgs.llvmPackages.libunwind` because `pkgs.libunwind` is too new to work with binary packaged by wrangler

thank you for clarification !

---

Also, `pkgs.llvmPackages.libraries.libcxx` is the same as `pkgs.llvmPackages.libcxx`, but I guess that doesn't matter.

---

thank you, i will keep that in mind

---

actually might as well make a change while here